src/main/java/org/dependencytrack/persistence/jdbi/ApiRequestStatementCustomizer.java

```diff
@@ -167,13 +166,11 @@ private void definePagination(final StatementContext ctx) {
     }
 
     private void defineProjectAclCondition(final StatementContext ctx) throws SQLException {
-        if (apiRequest == null) {
-            return;
-        }
-
-        if (apiRequest.getPrincipal() == null
+        if (apiRequest == null
+            || apiRequest.getPrincipal() == null
             || !isAclEnabled(ctx)
             || hasAccessManagementPermission(ctx, apiRequest.getPrincipal())) {
+            ctx.define(ATTRIBUTE_API_PROJECT_ACL_CONDITION, "TRUE");
```

Comment:
We always define this variable now, this way our queries don't have to check if it's set etc.